server/xcode/xcode_darwin.go

```diff
@@ -134,8 +136,19 @@ func (x *xcodeLocator) locate() error {
 			}
 		}
 	}
-	x.versions = versionMap
-	return nil
+	return versionMap
+}
+
+func xcodePlistForPath(path string) (*xcodePlist, error) {
+	var xcodePlist xcodePlist
+	versionFileReader, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open %s: %s", path, err.Error())
+	}
+	if err := plist.NewXMLDecoder(versionFileReader).Decode(&xcodePlist); err != nil {
+		return nil, fmt.Errorf("Failed to decode %s: %s", path, err.Error())
```

Comment:
In our codebase we prefer to use the errors from the server/util/status package rather than minting our own. They have some benefits:
 - map cleanly to grpc errors
 - can embed a stack trace if a config option is enabled
 - are easily "parseable" with Is*Error, or even into protos

So we could make these:
 >> return nil, status.InternalErrorf("Failed to open %s: %s", path, err.Error())

and:
 >> return nil, status.InternalErrorf("Failed to decode %s: %s", path, err.Error())